libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Steps/SwitchCondition.cs

```diff
@@ -74,30 +78,45 @@ protected override async Task<DialogTurnResult> OnRunCommandAsync(DialogContext
                 lock (this.Condition)
                 {
                     var engine = new ExpressionEngine();
-                    Expression condition = engine.Parse(this.Condition);
+                    conditionExpression = engine.Parse(this.Condition);
                     if (this.caseExpressions == null)
                     {
                         this.caseExpressions = new Dictionary<string, Expression>();
                         foreach (var c in this.Cases)
                         {
-                            var caseCondition = Expression.EqualsExpression(condition, engine.Parse(c.Value));
-                            // map of expression to steps
+                            // Values for cases are always coerced to string
+                            var caseCondition = Expression.ConstantExpression(c.Value);
```

Comment:
>var caseCondition = Expression.ConstantExpression(c.Value); [](start = 28, length = 59)

this doesn't seem right, why not this?

    var caseCondition = Expression.EqualsExpression(condition, Expression.ConstantExpression(c.Value));

???